pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -231,6 +231,11 @@ type CertificatePrivateKey struct {
 	// No other values are allowed.
 	// +optional
 	Size int `json:"size,omitempty"` // Validated by webhook. Be mindful of adding OpenAPI validation- see https://github.com/jetstack/cert-manager/issues/3644
+
+	// If `withDerFormar` is `true`, a new entry `tls.der` will be added in the secret,
+	// containing the binary value of the private key.
+	// +optional
+	WithDerFormat bool `json:"withDerFormat,omitempty"`
```

Comment:
Totally aside from whether we should or shouldn't add this feature, I think this field makes more sense as a `format` field or similar (rather than a bool option). This would also allow us to expand the supported options in future.

Reply:
Fully agree with this, if we're gonna add this functionality I'd propose something like:

```suggestion
	AdditionalKeyFormats []string `json:"additionalKeyFormats,omitempty"`
```

specified like this:

```yaml
  privateKey:
    algorithm: ECDSA
    size: 256
    additionalKeyFormats: ["tls.der"]
```

where each entry in the array maps to the key which would be added to the secret. This would make it pretty easy to add other formats for the private key.

Reply:
updated PR to add `additionalOutputFormats`.